### PR TITLE
Use GPIOXX in sample configs

### PR DESCRIPTION
**Pull request by clydebarrow**
## Description:

Use GPIOXX which is now detected during validation as per https://github.com/esphome/esphome/pull/6419

Also some spelling corrections and option description fixes.

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/touchscreen/cst226.rst [hunk 1]
```diff
@@ -27,18 +27,33 @@ Base Touchscreen Configuration
...
 - All other options from :ref:`Touchscreen <config-touchscreen>`.
 
 
+Sample config for the T4-S3
+***************************
+
+.. code-block:: yaml
+
+    i2c:
+      sda: 6
```
Comment (nielsnl68) [inline components/touchscreen/cst226.rst]:
```suggestion
      sda: GPIO6
```

BTW. should this not be **REPLACEME** as well? Or has this to do that this is a T4-S3 example.

File: components/touchscreen/cst226.rst [hunk 2]
```diff
@@ -27,18 +27,33 @@ Base Touchscreen Configuration
...
+      scl: 7
+    
+    touchscreen:
+      - platform: cst226
+        id: my_touchscreen
+        interrupt_pin: 8
+        reset_pin: 17
```
Comment (nielsnl68) [inline components/touchscreen/cst226.rst]:
```suggestion
        interrupt_pin: GPIO8
        reset_pin: GPIO17
```

**Review by nielsnl68 — APPROVED** [review]
Just one set of changes  adding **GPIO** before pin numbers.

File: components/touchscreen/cst226.rst (hunk 1 above)
Comment (clydebarrow) [reply components/touchscreen/cst226.rst]: This is a specific example for the T4-S3. The generic minimal config above has REPLACEME.

File: components/touchscreen/cst226.rst (hunk 2 above)
Comment (clydebarrow) [reply components/touchscreen/cst226.rst]: Is there a particular benefit in using the GPIOxx form?

Comment (nagyrobi) [reply components/touchscreen/cst226.rst]: You never know if it's a D8 or a GPIO8 lol...

Comment (clydebarrow) [reply components/touchscreen/cst226.rst]: Well, 8 always means GPIO8 and Dx is AFAIK only used on 8266, but I take your point - noobs might not know that.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**